application/controllers/admin/Quotas.php

```diff
@@ -344,7 +353,8 @@ public function delans(int $iSurveyId)
         if (empty($QuotaMember)) {
             throw new CHttpException(404, gT("Quota memeber not found."));
         }
-        $this->checkPermissions($iSurveyId, 'update', $QuotaMember->quota_id);
+        $this->checkPermissions($iSurveyId, 'update');
```

Comment:
This checkPermissions shouldn't be called using QuotaMemeber->sid ?

Reply:
> This checkPermissions shouldn't be called using QuotaMemeber->sid ?

Look at the new code :  i create a second function, and use it here.

```
$this->checkPermissions($iSurveyId, 'update');
$this->checkValidQuotaId($iSurveyId, $QuotaMember->quota_id);
```

Reply:
I meant the iSurveyId that is used for CheckPermission comes from parameter, not from the model.
I guess it is could be a bit safer... not sure how, but feels a bit more protected...

Reply:
Just disregard if you disagree.

Reply:
> I meant the iSurveyId that is used for CheckPermission comes from parameter, not from the model.

We are in 5.X : i don't disable usage of surveyId by url … i do it in master : remove usage of surveyId (in controller) when i can.

>  I guess it is could be a bit safer... not sure how, but feels a bit more protected...

in checkValidQuotaId : i control if $iSurveyId is the same on thequota with $QuotaMember->quota_id

Here : https://github.com/LimeSurvey/LimeSurvey/pull/3634/files#diff-6c267486bf17814384c0ac50ac57c00ce39fa9a3affd8dd2527f9275933f9e68R85

Reply:
ok

Reply:
:)